### PR TITLE
Feature/neo4j rebased

### PR DESCRIPTION
**Pull request by Dalamar42**

### CODE REVIEW

File: atlas-neo4j/src/main/java/org/atlasapi/neo4j/service/ContentNeo4jStore.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+                broadcastWriter,
+                locationWriter,
+                hierarchyWriter
+        );
+    }
+
+    public void writeEquivalences(ResourceRef subject, Set<ResourceRef> assertedAdjacents,
+            Set<Publisher> sources) {
+        try (Transaction transaction = sessionFactory.getSession().beginTransaction()) {
+            try {
```
Comment (emilssolmanis) [inline atlas-neo4j/src/main/java/org/atlasapi/neo4j/service/ContentNeo4jStore.java]: This is pointless, you can just have a catch on the resources one.

File: atlas-neo4j/src/main/java/org/atlasapi/neo4j/service/ContentNeo4jStore.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                transaction.failure();
+                Throwables.propagateIfInstanceOf(e, Neo4jPersistenceException.class);
+                throw Neo4jPersistenceException.create("Failed to write equivalences", e);
+            }
+        }
+    }
+
+    public void writeContent(Content content) {
+        try (Transaction transaction = sessionFactory.getSession().beginTransaction()) {
+            try {
```
Comment (emilssolmanis) [inline atlas-neo4j/src/main/java/org/atlasapi/neo4j/service/ContentNeo4jStore.java]: As above.

File: atlas-neo4j/src/main/java/org/atlasapi/neo4j/service/ContentNeo4jStore.java (hunk 1 above)
Comment (Dalamar42) [inline atlas-neo4j/src/main/java/org/atlasapi/neo4j/service/ContentNeo4jStore.java]: If I put the catch on the resources one the `transaction` variable is out of scope.

File: atlas-core/src/main/java/org/atlasapi/content/AbstractEquivalentContentStore.java [hunk 3]
```diff
@@ -167,19 +173,37 @@ private Content resolveId(Id contentId) throws WriteException {
...
+                graphUpdate.getAssertion() != null
+                ? getPartitionKey(graphUpdate.getAssertion().getSubject().getId())
+                : getPartitionKey(graphUpdate.getUpdated().getId())
+        );
+    }
+
     private void sendEquivalentContentChangedMessage(Content content, EquivalenceGraph graph)
             throws MessagingException {
-        equivalentContentUpdatedMessageSender.sendMessage(
+        contentUpdatedMessageSender.sendMessage(
```
Comment (emilssolmanis) [inline atlas-core/src/main/java/org/atlasapi/content/AbstractEquivalentContentStore.java]: Why do we send content change messages from here now? What's actually changed on the content..?

Comment (Dalamar42) [inline atlas-core/src/main/java/org/atlasapi/content/AbstractEquivalentContentStore.java]: @emilssolmanis It's the same variable. I just renamed it, but yeah that looks a bit misleading. I'll rename it back

File: atlas-neo4j/src/test/java/org/atlasapi/neo4j/service/writers/ContentWriterIT.java [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        writeContent(getContent(new Film(), 0L, Publisher.METABROADCAST));
+    }
+
+    @Test
+    public void writeSong() throws Exception {
+        writeContent(getContent(new Song(), 0L, Publisher.METABROADCAST));
+    }
+
+    @Test
+    public void writeEpisodeWithNoEpisodeNumber() throws Exception {
```
Comment (emilssolmanis) [inline atlas-neo4j/src/test/java/org/atlasapi/neo4j/service/writers/ContentWriterIT.java]: Bit of a nitpick but generally for test naming you want the expectation in the name as well, e.g., this one would be `writeEpisodeWithNoEpisodeNumber ... ` -- what? `.. throwsNullPointer`? `... succeeds`?

File: atlas-processing/src/main/java/org/atlasapi/messaging/Neo4jContentStoreContentUpdateWorker.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        return new Neo4jContentStoreContentUpdateWorker(
+                contentResolver, neo4JContentStore, timer, failureMeter
+        );
+    }
+
+    @Override
+    public void process(EquivalentContentUpdatedMessage message) throws RecoverableException {
+        Id contentId = message.getContentRef().getId();
+
+        log.debug("Processing message on id {}, took PT{}S, message: {}",
```
Comment (emilssolmanis) [inline atlas-processing/src/main/java/org/atlasapi/messaging/Neo4jContentStoreContentUpdateWorker.java]: You're not passing in the message even though you have a marker there. Think that might throw weird RTEs.

File: atlas-neo4j/src/test/java/org/atlasapi/neo4j/service/writers/ContentWriterIT.java (hunk 4 above)
Comment (Dalamar42) [inline atlas-neo4j/src/test/java/org/atlasapi/neo4j/service/writers/ContentWriterIT.java]: Ack. Fixing

File: atlas-processing/src/main/java/org/atlasapi/messaging/Neo4jContentStoreContentUpdateWorker.java (hunk 5 above)
Comment (Dalamar42) [inline atlas-processing/src/main/java/org/atlasapi/messaging/Neo4jContentStoreContentUpdateWorker.java]: Ack. Fixing

**Comment by emilssolmanis** [comment]
Generally lgtm, but we'll only really start getting the bugs by actually running it. 

ship it

**Pull request merged**